alerter.c

```diff
@@ -11,16 +18,32 @@ int networkAlertStub(float celcius) {
     return 200;
 }
 
+//fucntion for actual network alert
+int networkAlert(float celcius) {
+    printf("ALERT: Temperature is %.1f celcius.\n", celcius);
+    // Return 200 for ok
+    // Return 500 for not-ok
+    // stub always succeeds and returns 200
+    return 200;
+}
+
 void alertInCelcius(float farenheit) {
     float celcius = (farenheit - 32) * 5 / 9;
+#if ENV == TEST_ENV

```

Comment:
Using conditional compilation to execute test or production code increases complexity as the code size grows. Maintenance becomes difficult.
Test code and production code should be maintained separately.

Production code should not be aware of test environment or production envt.

Use dependency injection to solve this problem.

Reply:
![image](https://user-images.githubusercontent.com/70900516/206863992-86899dca-b7f4-4000-be43-ffa237b66969.png)